Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -0,0 +1,363 @@
+    input_vector[i].diov_buffer = (char *)into[i].data();
+    total += into[i].size();
+  }
+
+  ssize_t s = davixPosix->preadVec(m_fd, input_vector, output_vector, buffers, &davixErr);
+  if (davixErr) {
+    std::unique_ptr<DavixError> davixErrManaged(davixErr);
+    edm::Exception ex(edm::errors::FileReadError);
+    ex << "Davix::readv(name='" << m_name->c_str() << "', buffers=" << (buffers)
+       << ") failed with error " << davixErr->getErrMsg().c_str() << " and error code "
+       << davixErr->getStatus() << " and call returned " << s << " bytes";
+    ex.addContext("Calling DavixFile::readv()");
+    throw ex;
+  }
+  // Davix limits number of requests sent to the server
+  // to improve performance and it does range coalescing.
+  // So we can`t check what preadVec returns with what was requested.
+  // Example: If two ranges are overlapping, [10, 20] and [20, 30] which is
+  // coalesced into [10, 30] and it will contain one less byte than was requested.
+  // Only check if returned val <= 0 and make proper actions.
+  if (s < 0) {
+    edm::Exception ex(edm::errors::FileReadError);
+    ex << "Davix::readv(name='" << m_name->c_str() << "') failed and call returned " << s;
+    ex.addContext("Calling DavixFile::readv()");
+    throw ex;
+  } else if (s == 0) {
+    // end of file
+    return 0;
+  }
+  return total;
+}
+
+IOSize DavixFile::readv(IOPosBuffer *into, IOSize buffers) {
+  assert(!buffers || into);
+
+  // Davix does not support 0 buffers;
+  if (buffers == 0)
+    return 0;
+
+  DavixError *davixErr = NULL;
+
+  DavIOVecInput input_vector[buffers];
+  DavIOVecOuput output_vector[buffers];
+  IOSize total = 0;
+  for (IOSize i = 0; i < buffers; ++i) {
+    input_vector[i].diov_offset = into[i].offset();
+    input_vector[i].diov_size = into[i].size();
+    input_vector[i].diov_buffer = (char *)into[i].data();
+    total += into[i].size();
+  }
+  ssize_t s = davixPosix->preadVec(m_fd, input_vector, output_vector, buffers, &davixErr);
+  if (davixErr) {
+    std::unique_ptr<DavixError> davixErrManaged(davixErr);
+    edm::Exception ex(edm::errors::FileReadError);
+    ex << "Davix::readv(name='" << m_name->c_str() << "', n=" << buffers << ") failed with error "
+       << davixErr->getErrMsg().c_str() << " and error code " << davixErr->getStatus()
+       << " and call returned " << s << " bytes";
+    ex.addContext("Calling DavixFile::readv()");
+    throw ex;
+  }
+  // Davix limits number of requests sent to the server
+  // to improve performance and it does range coalescing.
+  // So we can`t check what preadVec returns with what was requested.
+  // Example: If two ranges are overlapping, [10, 20] and [20, 30] which is
+  // coalesced into [10, 30] and it will contain one less byte than was requested.
+  // Only check if returned val <= 0 and make proper actions.
+  if (s < 0) {
+    edm::Exception ex(edm::errors::FileReadError);
+    ex << "Davix::readv(name='" << m_name->c_str() << "', n=" << buffers
```

Comment:
No need for calling `m_name->c_str()` especially when `m_name` is no longer a `std::unique_ptr`.